addon/components/select-component.js

```diff
@@ -356,19 +360,31 @@ export default Ember.Component.extend(
     });
     var sortedGroupObjs = groupObjs.sort(this.get('groupSortFunction'));
     var result = Ember.A();
+    // If grouped header is in collapsedHeaders list, hide the child options by removing the child members from the
+    // backing object
+    // In order to hide items from the list, the object has to be removed from the backing object due to the use of
+    // vertical-collection's occlusion. If CSS were to be used to hide list items, there would not be a method to
+    // schedule vertical-collection's re-measure and re-render.
     sortedGroupObjs.forEach(function(groupObj) {
+      var isGroupHeaderCollapsed = collapsedHeaders.includes(groupObj.name);
       if (groupObj.name) {
         result.pushObject(
           Ember.Object.create({
             isGroupOption: true,
+            isGroupHeaderCollapsed: isGroupHeaderCollapsed,
             name: groupObj.name
           })
         );
       }
-      result.pushObjects(groupObj.members);
+      if (!isGroupHeaderCollapsed) {
+        result.pushObjects(groupObj.members);
+      }
     });
     return result;
-  }).property('preparedContent', 'preparedContent.[]', 'optionGroupPath', 'labels.[]', 'groupSortFunction'),
+  }).property('preparedContent', 'preparedContent.[]', 'optionGroupPath', 'labels.[]', 'groupSortFunction', 'collapsedGroupHeaders.[]'),
```

Comment:
Should this also observe `isGroupHeaderCollapsible`? And while we're at it, should we move these dependent keys to the beginning of the computed definition in line with newer code?

Reply:
I was thinking the same thing.  The prototype extensions feel quite dated, but perhaps this is just to follow along with existing style.